src/backend/EinsumNode.h

```diff
@@ -79,6 +80,26 @@ class einsum_ir::backend::EinsumNode {
     //! binary contraction
     BinaryContraction * m_cont = nullptr;
 
+    //! Memory manager for intermendiate results
+    MemoryManager * m_memory = nullptr;
+
+    //! id of allocated memoy
+    int64_t m_mem_id = 0;
+
+    //! the required number of frees until reserved memory is actualy freed
+    int64_t m_req_mem_frees = 1;
```

Comment:
The initial values of the values are all zero. Use the ``init`` function to provide a valid value